examples/spgemm/test_spgemm.cpp

```diff
@@ -0,0 +1,41 @@
+#include <fstream>
+#include <iostream>
+#include <loops/algorithms/spgemm/thread_mapped.cuh>
+
+using type_t = float;
+
+void copyDeviceMtxToHost(const loops::matrix_t<type_t, loops::memory_space_t::device>& d_C, loops::matrix_t<type_t, loops::memory_space_t::host>& h_C){
+    // Ensure the host matrix has the correct dimensions
+    h_C.rows = d_C.rows;
+    h_C.cols = d_C.cols;
+
+    // Allocate memory for the host matrix data
+    h_C.m_data.resize(d_C.rows * d_C.cols);
```

Comment:
You can see it here, what is the size of `d_C.rows * d_C.cols`? I think this is $N^2$ problem, which can easily exceed the total device memory a GPU has available.